lib/seahorn/BvOpSem2.cc

```diff
@@ -889,8 +889,16 @@ class OpSemMemManager {
   /// \brief Executes symbolic memcpy from physical memory with concrete length
   Expr MemCpy(PtrTy dPtr, char* sPtr, unsigned len, uint32_t align) {
     assert(len % align == 0);
-    WARN << "skipping MemCpy range";
-    return nullptr;
+    Expr res = m_ctx.read(m_ctx.getMemReadRegister());
+    for (int i = 0; i < len; i++) {
+      Expr offset = bv::bvnum(i, ptrSzInBits(), m_efac);
+      Expr dIdx = i == 0 ? dPtr : mk<BADD>(dPtr, offset);
+      Expr val = bv::bvnum(sPtr[i], ptrSzInBits(), m_efac);
```

Comment:
this writes an `8` bit value into `ptrSzInBits()` wide memory word. This is not quite what you want.
You want to read `sPtr` one semantic machine word at a time, create `bvnum`, as  you do, and store this.

In the current configuration, both pointers and machine words are `32` bits, so that's why the `store` succeeds.
But you are still updating memory incorrectly. If `sPtr` has one 32 bit word with octets `abcd`, you are writing `000a000b000c000d` into memory. Actually not quite due to correct computation of byte addresses.
Hope you get the picture